great_expectations/experimental/datasources/postgres_datasource.py

```diff
@@ -33,23 +31,44 @@ class BatchRequestError(Exception):
     pass
 
 
-# For our year splitter we default the range to the last 2 year.
-_CURRENT_YEAR = datetime.now(dateutil.tz.tzutc()).year
-_DEFAULT_YEAR_RANGE = list(range(_CURRENT_YEAR - 1, _CURRENT_YEAR + 1))
-_DEFAULT_MONTH_RANGE = list(range(1, 13))
-
-
 @pydantic_dc.dataclass(frozen=True)
 class ColumnSplitter:
-    method_name: str
     column_name: str
-    # param_defaults is a Dict where the keys are the parameters of the splitter and the values are the default
-    # values are the default values if a batch request using the splitter leaves the parameter unspecified.
-    param_defaults: Dict[str, List] = pydantic.Field(default_factory=dict)
+    method_name: str
+    param_names: List[str]
 
-    @property
-    def param_names(self) -> List[str]:
-        return list(self.param_defaults.keys())
+    def param_defaults(self, data_asset: DataAsset) -> Dict[str, List]:
+        raise NotImplementedError
+
+
+@pydantic_dc.dataclass(frozen=True)
+class SqlYearMonthSplitter(ColumnSplitter):
```

Comment:
This is the main change in this PR. We now query the database for the ranges over which to compute batches.